src/components/fx/helpers.js

```diff
@@ -83,3 +84,29 @@ function quadrature(dx, dy) {
         return Math.sqrt(x * x + y * y);
     };
 }
+
+/** Appends values inside array attributes corresponding to given point number
+ *
+ * @param {object} pointData : point data object (gets mutated here)
+ * @param {object} trace : full trace object
+ * @param {number} pointNumber : point number
+ */
+exports.appendArrayPointValue = function(pointData, trace, pointNumber) {
+    var arrayAttrs = trace._arrayAttrs;
+
+    for(var i = 0; i < arrayAttrs.length; i++) {
+        var astr = arrayAttrs[i];
+        var key;
+
+        if(astr === 'ids') key = 'id';
+        else if(astr === 'locations') key = 'location';
+        else key = astr;
+
+        if(pointData[key] === undefined) {
+            var val = Lib.nestedProperty(trace, astr).get();
```

Comment:
not blocking, but if we find that this starts dragging on performance (particularly for `selecting` I guess) then I suspect `nestedProperty` would be the primary cost. Could possibly store these instead of attribute strings in `_arrayAttrs`? It would be a little weird, since they could be referencing data arrays inside an outdated `trace` if we've redrawn without a recalc, but in as far as recalc is always triggered by changing an array attribute it *seems* like it would be ok?